lib/pf/provisioner/google_workspace_chromebook.pm

```diff
@@ -0,0 +1,465 @@
+            $logger->error("Unable to contact the Google API to import new devices.");
+            return $err;
+        }
+
+        my $chromeosdevices = $list->{chromeosdevices} // [];
+        $self->_import_devices($chromeosdevices);
+        $nextToken = $list->{nextPageToken};
+        last if !defined $nextToken;
+    }
+}
+
+sub importLastSyncQuery {
+    my ($self) = @_;
+    return "sync:0000-01-01..";
+}
+
+sub _import_devices {
+    my ($self, $chromeosdevices) = @_;
+    for my $device (@$chromeosdevices) {
+        $self->_import_device($device);
+    }
+}
+
+sub _import_device {
+    my ($self, $device) = @_;
+    return if $device->{status} ne 'ACTIVE';
+    my $role = $self->role_to_apply;
+    for my $f (qw(macAddress ethernetMacAddress ethernetMacAddress0)) {
+        next if !exists $device->{$f};
+        my $mac = $device->{$f};
+        next if !defined $mac;
+        node_register($mac, $default_pid , category => $role );
+    }
+}
+
+sub syncDateToQuery {
+    my ($date) = @_;
+    my ($ss,$mm,$hh,$day,$month,$year,$zone) = strptime($date);
+    return sprintf("sync:%04d-%02d-%02dT%02d:%02d:%02d..",$year+1900, $month+1,$day, $hh, $mm, $ss);
+}
+
+sub getDevice {
+    my ($self, $mac) = @_;
+    $mac =~ s/://g;
+    my ($err, $list) = $self->getList( { query => $mac } );
+    if (defined $err) {
+        $logger->error($list);
+        return $err, undef;
+    }
+
+    my $chromeosdevices = $list->{chromeosdevices} // [];
+    if (@$chromeosdevices) {
+        return undef, $chromeosdevices->[0];
+    }
+
+    return $FALSE, undef;
+}
+
+sub getList {
+    my ($self, $options) = @_;
+    my $url = $self->urlForList($options);
+    my $curl = $self->curl($url);
+    my $response_body;
+    $curl->setopt(CURLOPT_WRITEDATA, \$response_body);
+    my $curl_return_code = $curl->perform;
+    my $response;
+
+    # Looking at the results...
+    if ( $curl_return_code == 0 ) {
+        my $response_code = $curl->getinfo(CURLINFO_HTTP_CODE);
+        $response = decode_json($response_body);
+        if($response_code == 200) {
+            return undef, $response;
+        }
+
+        return $FALSE, $response;
+    }
+
+    my $msg = "An error occured while sending a JSONRPC request: $curl_return_code ".$curl->strerror($curl_return_code)." ".$curl->errbuf;
+    return -1, $msg;;
+}
+
+sub curl {
+    my ($self, $url) = @_;
+    my $response_body = '';
+    my $curl = WWW::Curl::Easy->new;
+    $curl->setopt(CURLOPT_HEADER, 0);
+    $curl->setopt(CURLOPT_URL, $url );
+    $curl->setopt(CURLOPT_NOSIGNAL, 1);
+    $curl->setopt(CURLOPT_NOPROGRESS, 1 );
+    $curl->setopt(CURLOPT_DNS_USE_GLOBAL_CACHE, 0);
+    return $curl;
+}
+
+sub process_device {
+    my ($self, $device) = @_;
+    for my $f (qw(macAddress ethernetMacAddress ethernetMacAddress0)) {
+        if (!exists $device->{$f}) {
+            next;
+        }
+
+        my $mac = clean_mac($device->{$f});
+        unless ($mac) {
+            next;
```

Comment:
We should log a warning saying that a MAC was found in field XYZ but doesn't match the expected format since this would imply Google starts using a MAC format we don't understand